From c58f1678795219f5fb89da251c9b9eeef5a71eb3 Mon Sep 17 00:00:00 2001
From: Bo Shen <voice.shen@atmel.com>
Date: Fri, 17 Oct 2014 14:45:05 +0800
Subject: [PATCH] ARM: atmel: sama5d4ek: add MACB1 support

As the MACB1 share pin with LCD, so enable MACB1 must disable LCD.
So, use sama5d4ek_2macb_mmc, sama5d4ek_2macb_nandflash,
sama5d4ek_2macb_spiflash to enable MACB1 and disable LCD.

(Need to close JP2 on EK board)

Signed-off-by: Bo Shen <voice.shen@atmel.com>
Acked-by: Nicolas Ferre <nicolas.ferre@atmel.com>
---
 board/atmel/sama5d4ek/sama5d4ek.c | 27 +++++++++++++++++++++++++++
 boards.cfg                        |  3 +++
 include/configs/sama5d4ek.h       |  2 ++
 3 files changed, 32 insertions(+)

diff --git a/board/atmel/sama5d4ek/sama5d4ek.c b/board/atmel/sama5d4ek/sama5d4ek.c
index c99643e..dc12e04 100644
--- a/board/atmel/sama5d4ek/sama5d4ek.c
+++ b/board/atmel/sama5d4ek/sama5d4ek.c
@@ -250,6 +250,25 @@ void sama5d4ek_macb0_hw_init(void)
 }
 #endif
 
+#ifdef CONFIG_MACB1
+void sama5d4ek_macb1_hw_init(void)
+{
+	at91_set_b_periph(AT91_PIO_PORTA, 2, 0);	/* TXCK */
+	at91_set_b_periph(AT91_PIO_PORTA, 4, 0);	/* TXEN */
+	at91_set_b_periph(AT91_PIO_PORTA, 14, 0);	/* TX0 */
+	at91_set_b_periph(AT91_PIO_PORTA, 15, 0);	/* Tx1 */
+	at91_set_b_periph(AT91_PIO_PORTA, 10, 0);	/* RXDV */
+	at91_set_b_periph(AT91_PIO_PORTA, 11, 0);	/* RXER */
+	at91_set_b_periph(AT91_PIO_PORTA, 12, 0);	/* RX0 */
+	at91_set_b_periph(AT91_PIO_PORTA, 13, 0);	/* RX1 */
+	at91_set_b_periph(AT91_PIO_PORTA, 23, 0);	/* MDIO */
+	at91_set_b_periph(AT91_PIO_PORTA, 22, 0);	/* MDC */
+
+	/* Enable clock */
+	at91_periph_clk_enable(ATMEL_ID_GMAC1);
+}
+#endif
+
 char *get_cpu_name()
 {
 	return "SAMA5D4";
@@ -294,9 +313,13 @@ int board_init(void)
 #ifdef CONFIG_MACB
 	sama5d4ek_macb0_hw_init();
 #endif
+#ifdef CONFIG_MACB1
+	sama5d4ek_macb1_hw_init();
+#else
 #ifdef CONFIG_LCD
 	sama5d4ek_lcd_hw_init();
 #endif
+#endif
 #ifdef CONFIG_CMD_USB
 	sama5d4ek_usb_hw_init();
 #endif
@@ -319,5 +342,9 @@ int board_eth_init(bd_t *bis)
 	rc = macb_eth_initialize(0, (void *)ATMEL_BASE_GMAC0, 0x00);
 #endif
 
+#ifdef CONFIG_MACB1
+	rc = macb_eth_initialize(1, (void *)ATMEL_BASE_GMAC1, 0x00);
+#endif
+
 	return rc;
 }
diff --git a/boards.cfg b/boards.cfg
index 359105c..9e4cf24 100644
--- a/boards.cfg
+++ b/boards.cfg
@@ -295,6 +295,9 @@ Active  arm         armv7          at91        atmel           sama5d4_xplained
 Active  arm         armv7          at91        atmel           sama5d4ek           sama5d4ek_mmc                         sama5d4ek:SAMA5D4,SYS_USE_MMC                                                                                                     Bo Shen <voice.shen@atmel.com>
 Active  arm         armv7          at91        atmel           sama5d4ek           sama5d4ek_nandflash                   sama5d4ek:SAMA5D4,SYS_USE_NANDFLASH                                                                                               Bo Shen <voice.shen@atmel.com>
 Active  arm         armv7          at91        atmel           sama5d4ek           sama5d4ek_spiflash                    sama5d4ek:SAMA5D4,SYS_USE_SERIALFLASH                                                                                             Bo Shen <voice.shen@atmel.com>
+Active  arm         armv7          at91        atmel           sama5d4ek           sama5d4ek_2macb_mmc                   sama5d4ek:SAMA5D4,MACB1,SYS_USE_MMC                                                                                               Bo Shen <voice.shen@atmel.com>
+Active  arm         armv7          at91        atmel           sama5d4ek           sama5d4ek_2macb_nandflash             sama5d4ek:SAMA5D4,MACB1,SYS_USE_NANDFLASH                                                                                         Bo Shen <voice.shen@atmel.com>
+Active  arm         armv7          at91        atmel           sama5d4ek           sama5d4ek_2macb_spiflash              sama5d4ek:SAMA5D4,MACB1,SYS_USE_SERIALFLASH                                                                                       Bo Shen <voice.shen@atmel.com>
 Active  arm         armv7          bcm281xx    broadcom        bcm28155_ap         bcm28155_ap                           bcm28155_ap                                                                                                                       Tim Kryger <tim.kryger@linaro.org>
 Active  arm         armv7          exynos      samsung         arndale             arndale                               -                                                                                                                                 Inderpal Singh <inderpal.singh@linaro.org>
 Active  arm         armv7          exynos      samsung         origen              origen                                -                                                                                                                                 Chander Kashyap <k.chander@samsung.com>
diff --git a/include/configs/sama5d4ek.h b/include/configs/sama5d4ek.h
index d50897e..d0a9731 100644
--- a/include/configs/sama5d4ek.h
+++ b/include/configs/sama5d4ek.h
@@ -134,6 +134,7 @@
 #define CONFIG_MACB_SEARCH_PHY
 
 /* LCD */
+#ifndef CONFIG_MACB1
 #define CONFIG_LCD
 #define LCD_BPP				LCD_COLOR16
 #define LCD_OUTPUT_BPP                  18
@@ -144,6 +145,7 @@
 #define CONFIG_ATMEL_HLCD
 #define CONFIG_ATMEL_LCD_RGB565
 #define CONFIG_SYS_CONSOLE_IS_IN_ENV
+#endif
 
 #ifdef CONFIG_SYS_USE_SERIALFLASH
 /* bootstrap + u-boot + env + linux in serial flash */
-- 
1.9.1

